src/main/java/school/lemon/changerequest/java/introduction/hw2/Task9.java

```diff
@@ -1,7 +1,43 @@
 package school.lemon.changerequest.java.introduction.hw2;
 
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class Task9 {
     public static String[] extractData(String URL) {
-        return new String[]{"", "", ""};
+        String[] arr = {"", "", ""};
+
+        Pattern patternSchema = Pattern.compile("([a-z]*://)");
+        Matcher matcherSchema = patternSchema.matcher(URL);
+        if (matcherSchema.find()) {
+            Pattern p = Pattern.compile("[a-z]*");
+            Matcher m = p.matcher(matcherSchema.group());
+            if (m.find()) {
+                arr[0] = m.group();
+            }
+        }
+
+        Pattern patternHost = Pattern.compile("([\\w]+\\.){1}([\\w]+\\.?)+");
+        Matcher matcherHost1 = patternHost.matcher(URL);
+        if (matcherHost1.find()) {
+            arr[1] = matcherHost1.group();
+        }
+
+        Pattern patternPort = Pattern.compile("(:[0-9]+)");
+        Matcher matcherPort = patternPort.matcher(URL);
+        if (matcherPort.find()) {
+            Pattern pattern11 = Pattern.compile("([0-9]+)");
+            Matcher matcher11 = pattern11.matcher(matcherPort.group());
+            if (matcher11.find()) {
+                arr[2] = matcher11.group();
+            }
+        }
+
+        return arr;
```

Comment:
You solution don't pass all test cases.
Also, This task could be solved using only 1 Pattern object and 1 Matcher object. There is no need for multiple Pattern-Matcher object.
Think about one complete regexp, that could cover all cases and data samples.